keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/probabilistic/MultinomialVertex.java

```diff
@@ -0,0 +1,55 @@
+package io.improbable.keanu.vertices.intgr.probabilistic;
+
+import static io.improbable.keanu.tensor.TensorShapeValidation.checkTensorsMatchNonScalarShapeOrAreScalar;
+
+import java.util.Map;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import io.improbable.keanu.distributions.discrete.Multinomial;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.update.ProbabilisticValueUpdater;
+
+public class MultinomialVertex extends IntegerVertex implements ProbabilisticInteger {
+
+    private final DoubleVertex p;
+    private final IntegerVertex n;
+
+    public MultinomialVertex(int[] tensorShape, IntegerVertex n, DoubleVertex p) {
+        super(new ProbabilisticValueUpdater<>());
+
+        int[] tensorShapePrefixedByOne = ArrayUtils.insert(0, tensorShape, 1);
+        checkTensorsMatchNonScalarShapeOrAreScalar(tensorShape, n.getShape());
+        checkTensorsMatchNonScalarShapeOrAreScalar(tensorShapePrefixedByOne, p.slice(0, 0).getShape());
```

Comment:
Is this just checking the length of p in dimension 1?

Reply:
No, it's checking the length of p in all the dimensions except 0.

I've changed it to this, which is clearer:

        int[] pShapeExcludingFirstDimension = ArrayUtils.remove(p.getShape(), 0);
        checkTensorsMatchNonScalarShapeOrAreScalar(tensorShape, pShapeExcludingFirstDimension);